ngmodules examples for final is not building

<!--
PLEASE HELP US PROCESS GITHUB ISSUES FASTER BY PROVIDING THE FOLLOWING INFORMATION.

ISSUES MISSING IMPORTANT INFORMATION MAY BE CLOSED WITHOUT INVESTIGATION.
-->

## I'm submitting a...
<!-- Check one of the following options with "x" -->
<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  <!-- Please search GitHub for a similar issue or PR before submitting -->
[ ] Feature request
[x] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
<!-- Describe how the issue manifests. -->
download The final version / download example on the page https://angular.io/guide/ngmodule#ngmodules and it will not build 

## Expected behavior
<!-- Describe what the desired behavior would be. -->
the project runs after you npm i > npm serve 

## Minimal reproduction of the problem with instructions
<!--
For bug reports please provide the *STEPS TO REPRODUCE* and if possible a *MINIMAL DEMO* of the problem via
https://plnkr.co or similar (you can use this template as a starting point: http://plnkr.co/edit/tpl:AvJOMERrnz94ekVua0u5).
-->
download the code and run it
## What is the motivation / use case for changing the behavior?
<!-- Describe the motivation or the concrete use case. -->
want people to get working code 

## Environment
prod
<pre><code>
Angular version: 5.0.0
<!-- Check whether this is still an issue in the most recent Angular version -->

Browser:
- [x ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: XX  <!-- run `node --version` -->
- Platform:  <!-- Mac, Linux, Windows -->
windows 10 pro - node newest - npm newest - angular/cli newest
Others:
<!-- Anything else relevant?  Operating system version, IDE, package manager, HTTP server, ... -->
</code></pre>
**first error**


ENOENT: no such file or directory, stat 'C:\dev\source\googleAngular\ngmodule\src\tsconfig.app.json'
Error: ENOENT: no such file or directory, stat 'C:\dev\source\googleAngular\ngmodule\src\tsconfig.app.json'
    at Object.fs.statSync (fs.js:948:11)
    at AngularCompilerPlugin._setupOptions (C:\dev\source\googleAngular\ngmodule\node_modules\@ngtools\webpack\src\angular_compiler_plugin.js:78:16)
    at new AngularCompilerPlugin (C:\dev\source\googleAngular\ngmodule\node_modules\@ngtools\webpack\src\angular_compiler_plugin.js:43:14)
    at _createAotPlugin (C:\dev\source\googleAngular\ngmodule\node_modules\@angular\cli\models\webpack-configs\typescript.js:77:16)
    at Object.getNonAotConfig (C:\dev\source\googleAngular\ngmodule\node_modules\@angular\cli\models\webpack-configs\typescript.js:100:19)
    at NgCliWebpackConfig.buildConfig (C:\dev\source\googleAngular\ngmodule\node_modules\@angular\cli\models\webpack-config.js:33:37)
    at Class.run (C:\dev\source\googleAngular\ngmodule\node_modules\@angular\cli\tasks\serve.js:71:98)
    at check_port_1.checkPort.then.port (C:\dev\source\googleAngular\ngmodule\node_modules\@angular\cli\commands\serve.js:123:26)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
PS C:\dev\source\googleAngular\ngmodule> ng serve --port 4100

added a basic version 
`{
    "extends": "../tsconfig.json",
    "compilerOptions": {
      "outDir": "../out-tsc/app",
      "module": "es2015",
      "baseUrl": "",
      "types": []
    },
    "exclude": [
      "test.ts",
      "**/*.spec.ts"
    ]
  }
  `

it then fails
`
ERROR in multi ./src/polyfills.ts
Module not found: Error: Can't resolve 'C:\dev\source\googleAngular\ngmodule\src\polyfills.ts' in 'C:\dev\source\googleAngular\ngmodule\node_modules\@angular\cli\models\webpack-configs'
resolve 'C:\dev\source\googleAngular\ngmodule\src\polyfills.ts' in 'C:\dev\source\googleAngular\ngmodule\node_modules\@angular\cli\models\webpack-configs'
  using description file: C:\dev\source\googleAngular\ngmodule\node_modules\@angular\cli\package.json (relative path: ./models/webpack-configs)
    Field 'browser' doesn't contain a valid alias configuration
  after using description file: C:\dev\source\googleAngular\ngmodule\node_modules\@angular\cli\package.json (relative path: ./models/webpack-configs)
    using description file: C:\dev\source\googleAngular\ngmodule\package.json (relative path: ./src/polyfills.ts)
      no extension
        Field 'browser' doesn't contain a valid alias configuration
        C:\dev\source\googleAngular\ngmodule\src\polyfills.ts doesn't exist
      .ts
        Field 'browser' doesn't contain a valid alias configuration
        C:\dev\source\googleAngular\ngmodule\src\polyfills.ts.ts doesn't exist
      .js
        Field 'browser' doesn't contain a valid alias configuration
        C:\dev\source\googleAngular\ngmodule\src\polyfills.ts.js doesn't exist
      as directory
        C:\dev\source\googleAngular\ngmodule\src\polyfills.ts doesn't exist
[C:\dev\source\googleAngular\ngmodule\src\polyfills.ts]
[C:\dev\source\googleAngular\ngmodule\src\polyfills.ts.ts]
[C:\dev\source\googleAngular\ngmodule\src\polyfills.ts.js]
[C:\dev\source\googleAngular\ngmodule\src\polyfills.ts]
 @ multi ./src/polyfills.ts
ERROR in TypeError: Cannot read property 'length' of undefined
    at createSourceFile (C:\dev\source\googleAngular\ngmodule\node_modules\typescript\lib\typescript.js:16640:110)
    at parseSourceFileWorker (C:\dev\source\googleAngular\ngmodule\node_modules\typescript\lib\typescript.js:16572:26)
    at Object.parseSourceFile (C:\dev\source\googleAngular\ngmodule\node_modules\typescript\lib\typescript.js:16521:26)
    at Object.createSourceFile (C:\dev\source\googleAngular\ngmodule\node_modules\typescript\lib\typescript.js:16371:29)
    at VirtualFileStats.getSourceFile (C:\dev\source\googleAngular\ngmodule\node_modules\@ngtools\webpack\src\compiler_host.js:67:35)
    at WebpackCompilerHost.getSourceFile (C:\dev\source\googleAngular\ngmodule\node_modules\@ngtools\webpack\src\compiler_host.js:207:38)
    at findSourceFile (C:\dev\source\googleAngular\ngmodule\node_modules\typescript\lib\typescript.js:70246:29)
    at processImportedModules (C:\dev\source\googleAngular\ngmodule\node_modules\typescript\lib\typescript.js:70393:25)
    at findSourceFile (C:\dev\source\googleAngular\ngmodule\node_modules\typescript\lib\typescript.js:70274:17)
    at args (C:\dev\source\googleAngular\ngmodule\node_modules\typescript\lib\typescript.js:70200:85)
    at getSourceFileFromReferenceWorker (C:\dev\source\googleAngular\ngmodule\node_modules\typescript\lib\typescript.js:70173:34)
    at processSourceFile (C:\dev\source\googleAngular\ngmodule\node_modules\typescript\lib\typescript.js:70200:13)
    at processRootFile (C:\dev\source\googleAngular\ngmodule\node_modules\typescript\lib\typescript.js:70055:13)
    at C:\dev\source\googleAngular\ngmodule\node_modules\typescript\lib\typescript.js:69312:60
    at Object.forEach (C:\dev\source\googleAngular\ngmodule\node_modules\typescript\lib\typescript.js:1506:30)
    at Object.createProgram (C:\dev\source\googleAngular\ngmodule\node_modules\typescript\lib\typescript.js:69312:16)

--- a/aio/content/examples/dependency-injection/plnkr.json
+++ b/aio/content/examples/dependency-injection/plnkr.json
@@ -5,7 +5,7 @@
     "!**/*.d.ts",
     "!**/*.js",
     "!**/*.[0,1,2].*",
-    "**/dummy.module.ts"
+    "!**/dummy.module.ts"
   ],
   "tags": ["dependency", "di"]
 }

--- a/aio/content/examples/ngmodule/src/index.0.html
+++ b/aio/content/examples/ngmodule/src/index.0.html
@@ -5,17 +5,6 @@
     <title>NgModule Minimal</title>
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1">
-    <link rel="stylesheet" href="styles.css">
-
-     <!-- Polyfills -->
-    <script src="node_modules/core-js/client/shim.min.js"></script>
-
-    <script src="node_modules/zone.js/dist/zone.js"></script>
-    <script src="node_modules/systemjs/dist/system.src.js"></script>
-    <script src="systemjs.config.js"></script>
-    <script>
-      System.import('main.0.js').catch(function(err){ console.error(err); });
-    </script>
   </head>
 
   <body>

--- a/aio/content/examples/ngmodule/src/index.1.html
+++ b/aio/content/examples/ngmodule/src/index.1.html
@@ -5,17 +5,6 @@
     <title>NgModule Minimal</title>
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1">
-    <link rel="stylesheet" href="styles.css">
-
-     <!-- Polyfills -->
-    <script src="node_modules/core-js/client/shim.min.js"></script>
-
-    <script src="node_modules/zone.js/dist/zone.js"></script>
-    <script src="node_modules/systemjs/dist/system.src.js"></script>
-    <script src="systemjs.config.js"></script>
-    <script>
-      System.import('main.1.js').catch(function(err){ console.error(err); });
-    </script>
   </head>
 
   <body>

--- a/aio/content/examples/ngmodule/src/index.1b.html
+++ b/aio/content/examples/ngmodule/src/index.1b.html
@@ -5,17 +5,6 @@
     <title>NgModule Minimal</title>
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1">
-    <link rel="stylesheet" href="styles.css">
-
-     <!-- Polyfills -->
-    <script src="node_modules/core-js/client/shim.min.js"></script>
-
-    <script src="node_modules/zone.js/dist/zone.js"></script>
-    <script src="node_modules/systemjs/dist/system.src.js"></script>
-    <script src="systemjs.config.js"></script>
-    <script>
-      System.import('main.1b.js').catch(function(err){ console.error(err); });
-    </script>
   </head>
 
   <body>

--- a/aio/content/examples/ngmodule/src/index.2.html
+++ b/aio/content/examples/ngmodule/src/index.2.html
@@ -5,17 +5,6 @@
     <title>NgModule Minimal</title>
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1">
-    <link rel="stylesheet" href="styles.css">
-
-     <!-- Polyfills -->
-    <script src="node_modules/core-js/client/shim.min.js"></script>
-
-    <script src="node_modules/zone.js/dist/zone.js"></script>
-    <script src="node_modules/systemjs/dist/system.src.js"></script>
-    <script src="systemjs.config.js"></script>
-    <script>
-      System.import('main.2.js').catch(function(err){ console.error(err); });
-    </script>
   </head>
 
   <body>

--- a/aio/content/examples/ngmodule/src/index.3.html
+++ b/aio/content/examples/ngmodule/src/index.3.html
@@ -5,17 +5,6 @@
     <title>NgModule Minimal</title>
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1">
-    <link rel="stylesheet" href="styles.css">
-
-     <!-- Polyfills -->
-    <script src="node_modules/core-js/client/shim.min.js"></script>
-
-    <script src="node_modules/zone.js/dist/zone.js"></script>
-    <script src="node_modules/systemjs/dist/system.src.js"></script>
-    <script src="systemjs.config.js"></script>
-    <script>
-      System.import('main.3.js').catch(function(err){ console.error(err); });
-    </script>
   </head>
 
   <body>

--- a/aio/content/examples/reactive-forms/final.plnkr.json
+++ b/aio/content/examples/reactive-forms/final.plnkr.json
@@ -5,13 +5,17 @@
     "styles.css",
 
     "app/app.component.ts",
+    "app/app.component.html",
+    "app/app.component.css",
     "app/app.module.ts",
     "app/data-model.ts",
     "app/hero.service.ts",
-    "app/hero-detail.component.html",
-    "app/hero-detail.component.ts",
-    "app/hero-list.component.html",
-    "app/hero-list.component.ts",
+    "app/hero-detail/hero-detail.component.html",
+    "app/hero-detail/hero-detail.component.ts",
+    "app/hero-detail/hero-detail.component.css",
+    "app/hero-list/hero-list.component.html",
+    "app/hero-list/hero-list.component.ts",
+    "app/hero-list/hero-list.component.css",
 
     "main-final.ts",
     "index-final.html"

--- a/aio/content/examples/reactive-forms/src/index-final.html
+++ b/aio/content/examples/reactive-forms/src/index-final.html
@@ -6,18 +6,6 @@
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1">
     <link rel="stylesheet" href="https://unpkg.com/bootstrap@3.3.7/dist/css/bootstrap.min.css">
-    <link rel="stylesheet" href="styles.css">
-
-     <script src="node_modules/core-js/client/shim.min.js"></script>
-
-    <script src="node_modules/zone.js/dist/zone.js"></script>
-    <script src="node_modules/reflect-metadata/Reflect.js"></script>
-    <script src="node_modules/systemjs/dist/system.src.js"></script>
-
-    <script src="systemjs.config.js"></script>
-    <script>
-      System.import('main-final.js').catch(function(err){ console.error(err); });
-    </script>
   </head>
 
   <body>

--- a/aio/tools/example-zipper/customizer/package-json/cli.json
+++ b/aio/tools/example-zipper/customizer/package-json/cli.json
@@ -1,6 +1,7 @@
 {
   "scripts": [
     { "name": "ng", "command": "ng" },
+    { "name": "build", "command": "ng build" },
     { "name": "start", "command": "ng serve" },
     { "name": "test", "command": "ng test" },
     { "name": "lint", "command": "ng lint" },

--- a/aio/tools/example-zipper/exampleZipper.js
+++ b/aio/tools/example-zipper/exampleZipper.js
@@ -54,6 +54,19 @@ class ExampleZipper {
     }
   }
 
+  // rename a custom main.ts or index.html file
+  _renameFile(file) {
+    if (/src\/main[-.]\w+\.ts$/.test(file)) {
+      return 'src/main.ts';
+    }
+
+    if (/src\/index[-.]\w+\.html$/.test(file)) {
+      return 'src/index.html';
+    }
+
+    return file;
+  }
+
   _zipExample(configFileName, sourceDirName, outputDirName) {
     let json = require(configFileName, 'utf-8');
     const basePath = json.basePath || '';
@@ -79,12 +92,16 @@ class ExampleZipper {
       'tslint.json',
       'karma-test-shim.js',
       'karma.conf.js',
+      'tsconfig.json',
       'src/testing/**/*',
       'src/.babelrc',
       'src/favicon.ico',
-      'src/typings.d.ts'
+      'src/polyfills.ts',
+      'src/typings.d.ts',
+      'src/environments/**/*',
+      'src/tsconfig.*'
     ];
-    var defaultExcludes = [
+    var alwaysExcludes = [
       '!**/bs-config.e2e.json',
       '!**/*plnkr.*',
       '!**/*zipper.*',
@@ -132,13 +149,14 @@ class ExampleZipper {
       }
     });
 
-    Array.prototype.push.apply(gpaths, defaultExcludes);
+    Array.prototype.push.apply(gpaths, alwaysExcludes);
 
     let fileNames = globby.sync(gpaths, { ignore: ['**/node_modules/**']});
 
     let zip = this._createZipArchive(outputFileName);
     fileNames.forEach((fileName) => {
       let relativePath = path.relative(exampleDirName, fileName);
+      relativePath = this._renameFile(relativePath);
       let content = fs.readFileSync(fileName, 'utf8');
       let extn = path.extname(fileName).substr(1);
       // if we don't need to clean up the file then we can do the following.

